src/pkg/packager/prepare.go

```diff
@@ -356,6 +377,28 @@ func (p *Packager) processUnstructuredImages(resource *unstructured.Unstructured
 	return matchedImages, maybeImages, nil
 }
 
+func (p *Packager) findWhyResources(resources []*unstructured.Unstructured, whyImage, componentName, resourceName string, isChart bool) []string {
+	output := []string{}
+	for _, resource := range resources {
+		bytes, _ := resource.MarshalJSON()
+		json := string(bytes)
+		resourceTypeKey := "manifest"
+		if isChart {
+			resourceTypeKey = "chart"
+		}
+
+		var imageFuzzyCheck = regexp.MustCompile(`(?mi)["|=]([a-z0-9\-.\/:]+:[\w.\-]*[a-z\.\-][\w.\-]*)"`)
+		matches := imageFuzzyCheck.FindAllStringSubmatch(json, -1)
+		for _, group := range matches {
+			if whyImage == group[1] {
```

Comment:
Since we already have the k8s resource and the why image as strings, we can simplify this further to use `strings.Contains` to check if the why image is present in the k8s resource